specification/nginx/resource-manager/NGINX.NGINXPLUS/stable/2023-09-01/swagger.json

```diff
@@ -0,0 +1,1579 @@
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/NginxConfiguration"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/NginxConfiguration"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Configurations_CreateOrUpdate": {
+            "$ref": "./examples/Configurations_CreateOrUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "NginxConfiguration"
+        ],
+        "summary": "Reset the NGINX configuration of given NGINX deployment to default",
+        "operationId": "Configurations_Delete",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeploymentNameParameter"
+          },
+          {
+            "in": "path",
+            "name": "configurationName",
+            "description": "The name of configuration, only 'default' is supported value due to the singleton of NGINX conf",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "202": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "Success"
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/ResourceProviderDefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Configurations_Delete": {
+            "$ref": "./examples/Configurations_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Nginx.NginxPlus/nginxDeployments/{deploymentName}/configurations/{configurationName}/analysis": {
```

Comment:
>analysis [](http://example.com/codeflow?start=167&length=8)

This needs to be a verb, e.g. analyze.

Reply:
It does seem like "analysis" should be "analyze" especially since the summary starts with "Analyze an NGINX configuration..." so the URL isn't consistent with its own documentation.
So, I agree this should be a noun.

If a version of the service has already shipped with "analysis" then changing it is a breaking change and we should not do it if any 3rd party customers could be using it. If the only customers are 1st party (like the Azure Portal), then we don't care about breaking 1st party as we can work with them closely to fix things without breaking customers and we always compromise 1st party to make a better long-term customer (3rd party) experience.

Also, we almost always compromise immediate negative business impact in favor of the long-term positive customer experience. If we could fix this now by introducing some delay (negative business impact) but not have to break customers in the future, then we should definitely do this. Breaking customers causes severe customer dissatisfaction with Azure all up - the breaking change board was created a few years ago within the Azure org with the sole charter of reducing all the breaks to improve overall Azure customer satisfaction.